vcf_annotation_tools/vcf_expression_annotator.py

```diff
@@ -104,14 +104,21 @@ def create_vcf_writer(args, vcf_reader):
         output_file = args.output_vcf
     return vcfpy.Writer.from_path(output_file, new_header)
 
-def add_expressions(entry, is_multi_sample, sample_name, df, items, tag, id_column, expression_column):
+def add_expressions(entry, is_multi_sample, sample_name, df, items, tag, id_column, expression_column, ignore_transcript_version):
     expressions = {}
     for item in items:
-        subset = df.loc[df[id_column] == item]
+        if tag == 'TX' and ignore_transcript_version:
+            df['transcript_without_version'] = df[id_column].apply(lambda x: x.split('.')[0])
+            subset = df.loc[df['transcript_without_version'] == item.split('.')[0]]
+        else:
```

Comment:
There are lots of different transcript databases (especially when you start getting into other species). It would be good to explicitly strip the last period and number, rather than assuming that the first portion of the transcript name doesn't contain periods. (this would do the wrong thing if the transcript was ASDF.ASDF.2)

Reply:
Is the last number always a single digit or can it be longer?

Reply:
It can be longer, IIRC.  (/\.\d+$/ would be the right regex)

Reply:
yup, its an integer that can eventually exceed 9.